workshop/4-Messaging/4.1-Pinpoint.ipynb

```diff
@@ -229,215 +233,13 @@
-    "                user_personalization_campaign_arn = campaign['campaignArn']\n",
-    "                print(f'Found existing campaign for solution: {user_personalization_campaign_arn}')\n",
-    "                \n",
-    "                response = personalize.describe_campaign(campaignArn = user_personalization_campaign_arn)\n",
-    "                if response['campaign']['solutionVersionArn'] == user_personalization_solution_arn:\n",
-    "                    print('Campaign is already using the latest solution version')\n",
-    "                else:\n",
-    "                    print('Updating campaign with the latest solution version')\n",
-    "                    response = personalize.update_campaign(\n",
-    "                        campaignArn = user_personalization_campaign_arn,\n",
-    "                        solutionVersionArn = user_personalization_solution_arn,\n",
-    "                        minProvisionedTPS = 1\n",
-    "                    )\n",
-    "                    print(json.dumps(response, indent=2))\n",
-    "                break"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "%%time\n",
-    "\n",
-    "campaign_arns = [ user_personalization_campaign_arn ]\n",
-    "\n",
-    "max_time = time.time() + 3*60*60 # 3 hours\n",
-    "while time.time() < max_time:\n",
-    "    for campaign_arn in reversed(campaign_arns):\n",
-    "        campaign_response = personalize.describe_campaign(\n",
-    "            campaignArn = campaign_arn\n",
-    "        )\n",
-    "        status = campaign_response[\"campaign\"][\"status\"]\n",
-    "        if status == 'ACTIVE' and campaign_response.get('latestCampaignUpdate'):\n",
-    "            status = campaign_response['latestCampaignUpdate']['status']\n",
-    "\n",
-    "        if status == \"ACTIVE\":\n",
-    "            print(f'Campaign {campaign_arn} successfully completed')\n",
-    "            campaign_arns.remove(campaign_arn)\n",
-    "        elif status == \"CREATE FAILED\":\n",
-    "            print(f'Campaign {campaign_arn} failed')\n",
-    "            if campaign_response.get('failureReason'):\n",
-    "                print('   Reason: ' + campaign_response['failureReason'])\n",
-    "            campaign_arns.remove(campaign_arn)\n",
-    "\n",
-    "    if len(campaign_arns) > 0:\n",
-    "        print('At least one campaign is still in progress')\n",
-    "        time.sleep(60)\n",
-    "    else:\n",
-    "        print(\"All campaigns have completed\")\n",
-    "        break"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Great! Now we have the Personalize campaign ARN that we need to create the Pinpoint ML recommender. Let's proceed to the next step."
-   ]
-  },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "### Get Recommendation Customizer Lambda ARN\n",
```

Comment:
Change to "Get Pinpoint campaign hook Lambda ARN" (or similar).

Also, the IAM role cell below is probably not necessary anymore since the Recommendations service is being called.

Reply:
Updated workshop as per comments in latest commit